tools/sklearn/lightgbm.xml

```diff
@@ -0,0 +1,291 @@
+<tool id="sklearn_lightgbm" name="LightGBM" version="@VERSION@">
+    <description>- train and apply LightGBM models</description>
+    <macros>
+        <import>main_macros.xml</import>
+    </macros>
+    <expand macro="python_requirements">
+          <requirement type="package" version="2.3.0">lightgbm</requirement>
+    </expand>
+    <expand macro="macro_stdio"/>
+    <version_command>echo "@VERSION@"</version_command>
+    <command><![CDATA[
+    python '$lightgbm_script' '$inputs'
+]]>
+    </command>
+    <configfiles>
+        <inputs name="inputs"/>
+        <configfile name="lightgbm_script">
+<![CDATA[
+import json
+import lightgbm as lgb
+import numpy as np
+import pandas
+import sys
+
+from scipy.io import mmread
+from galaxy_ml.utils import load_model, get_X_y
+
+
+N_JOBS = int(__import__('os').environ.get('GALAXY_SLOTS', 1))
+
+# Get inputs, outputs.
+input_json_path = sys.argv[1]
+with open(input_json_path, "r") as param_handler:
+    params = json.load(param_handler)
+print(params)
+
+# Put all cheetah up here to avoid confusion.
+#if $selected_tasks.selected_task == "train":
+infile1 = "$selected_tasks.selected_algorithms.input_options.infile1"
+infile2 = "$selected_tasks.selected_algorithms.input_options.infile2"
+#else:
+infile_model = "$selected_tasks.infile_model"
+infile_data = "$selected_tasks.infile_data"
+#end if
+outfile_fit = "$outfile_fit"
+outfile_predict = "$outfile_predict"
+
+# All Python from here on out:
+
+if params["selected_tasks"]["selected_task"] == "train":
+    algorithm = params["selected_tasks"]["selected_algorithms"]["selected_algorithm"]
+    options = params["selected_tasks"]["selected_algorithms"]["options"]
+    if "max_leaf_nodes" in options:
+        options["num_leaves"] = options["max_leaf_nodes"]
+        options.pop("max_leaf_nodes")
+
+    X, y = get_X_y(params, infile1, infile2)
+    lgb_train = lgb.Dataset(X, y)
+    gbm = lgb.train(options, lgb_train)
```

Comment:
It seems these estimators don't comply with sklearn `fit`, `predict` APIs. Wrapper is needed to make them work in pipeline.